pkg/kubelet/dockertools/manager.go

```diff
@@ -213,24 +216,25 @@ func NewDockerManager(
 	}
 
 	dm := &DockerManager{
-		client:                 client,
-		recorder:               recorder,
-		containerRefManager:    containerRefManager,
-		os:                     osInterface,
-		machineInfo:            machineInfo,
-		podInfraContainerImage: podInfraContainerImage,
-		dockerPuller:           newDockerPuller(client, qps, burst),
-		dockerRoot:             dockerRoot,
-		containerLogsDir:       containerLogsDir,
-		networkPlugin:          networkPlugin,
-		livenessManager:        livenessManager,
-		runtimeHelper:          runtimeHelper,
-		execHandler:            execHandler,
-		oomAdjuster:            oomAdjuster,
-		procFs:                 procFs,
-		cpuCFSQuota:            cpuCFSQuota,
-		enableCustomMetrics:    enableCustomMetrics,
-		configureHairpinMode:   hairpinMode,
+		client:                           client,
+		recorder:                         recorder,
+		containerRefManager:              containerRefManager,
+		os:                               osInterface,
+		machineInfo:                      machineInfo,
+		podInfraContainerImage:           podInfraContainerImage,
```

Comment:
Why not combine this and the secret into a single object. IIRC, there is an ImageSpec object which can be used here.